traffic_portal/app/src/common/modules/table/profilesParamsCompare/TableProfilesParamsCompareController.js

```diff
@@ -0,0 +1,213 @@
+			serverService.getServers({ profileId: profile.id }).
+				then(function(result) {
+					this['profile' + profNum + 'Usage'] = result.length + ' servers';
+				});
+		}
+	};
+
+	let confirmUpdateProfiles = function() {
+		// ok, this method is fun :)
+		let params = {
+			title: 'Modify ' + profile1.name + ' parameters',
+			message: 'The ' + profile1.name + ' profile is used by ' + this.profile1Usage + '. Are you sure you want to modify the parameters?'
+		};
+		let modalInstance = $uibModal.open({
+			templateUrl: 'common/modules/dialog/confirm/dialog.confirm.tpl.html',
+			controller: 'DialogConfirmController',
+			size: 'md',
+			resolve: {
+				params: function () {
+					return params;
+				}
+			}
+		});
+		modalInstance.result
+			.then(
+				function() { updateProfile1 = true; },
+				function() { updateProfile1 = false; }
+			)
+			.finally(
+				function() {
+					let params = {
+						title: 'Modify ' + profile2.name + ' parameters',
+						message: 'The ' + profile2.name + ' profile is used by ' + this.profile2Usage + '. Are you sure you want to modify the parameters?'
+					};
+					let modalInstance = $uibModal.open({
+						templateUrl: 'common/modules/dialog/confirm/dialog.confirm.tpl.html',
+						controller: 'DialogConfirmController',
+						size: 'md',
+						resolve: {
+							params: function () {
+								return params;
+							}
+						}
+					});
+					modalInstance.result
+						.then(
+							function() { updateProfile2 = true; },
+							function() { updateProfile2 = false; }
+						)
+						.finally(
+							function() {
+								let promises = [];
+
+								if (updateProfile1) {
+									promises.push(profileParameterService.linkProfileParameters(profile1, _.pluck($scope.selectedProfile1Params, 'id')));
+								}
+
+								if (updateProfile2) {
+									promises.push(profileParameterService.linkProfileParameters(profile2, _.pluck($scope.selectedProfile2Params, 'id')));
+								}
+
+								if (promises.length > 0) {
+									$q.all(promises)
+										.then(
+											function(result) {
+												let messages = [];
+												for (let i = 0; i < result.length; i++) {
+													messages = _.union(messages, result[i].data.alerts);
+												};
+												messageModel.setMessages(messages, false);
+											})
+										.finally(
+											function() {
+												$scope.refresh();
+											}
+										);
+								}
+							}
+						);
+				}
+			);
+	};
+
+	$scope.dirty = false;
+
+	$scope.profile1 = profile1;
+	$scope.profile2 = profile2;
+	$scope.profilesParams = profilesParams;
+
+	$scope.selectedProfile1Params = [];
+	$scope.selectedProfile2Params = [];
+
+	$scope.selectedParams = _.map(profilesParams, function(param) {
+		let isAssignedToProfile1 = _.find(profile1.params, function(profile1param) { return profile1param.id == param.id }),
+			isAssignedToProfile2 = _.find(profile2.params, function(profile2param) { return profile2param.id == param.id });
```

Comment:
I don't think `_.find` does what you're looking for. It returns the first value in the array that yields a truthy value from the predicate function. I think what you want to know is whether or not such a value *exists* - you don't care about the value. Instead, consider using [the built-in `Array.prototype.some` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) like so:
```javascript
// also note the use of triple-equals (===)
const isAssignedToProfile1 = profile1.params.some(function(p){return p.id === param.id});
//...
```

Reply:
it should have the same effect but i don't mind getting rid of the underscore method.

Reply:
there should be an issue for that, you wanna open one?

Reply:
https://github.com/apache/trafficcontrol/issues/3928